src/test/java/com/imagination/cbs/service/impl/EmailServiceImplTest.java

```diff
@@ -0,0 +1,335 @@
+
+	private Template template;
+
+	@Before
+	public void init() {
+		template = mock(Template.class);
+	}
+	
+	@Test
+	public void shouldSendEmailForBookingApproval() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException, MessagingException {
+
+		MailRequest emailRequest = createMailRequest();
+		BookingRevision bookingRevision = createBookingRevision(false);
+		ApprovalStatusDm approvalStatusDm = new ApprovalStatusDm();
+		approvalStatusDm.setApprovalName("Akshay");
+		ReflectionTestUtils.setField(emailServiceImpl, "baseUrl", "https://imaginagtion.com/bookingSummary?bookingId={bookingId}&status={status}");
+		
+		when(config.getTemplate("email.approval_request.ftl")).thenReturn(template);
+		when(approvalStatusDmRepository.findById(201L)).thenReturn(Optional.of(approvalStatusDm));
+		
+		emailServiceImpl.sendEmailForBookingApproval(emailRequest, bookingRevision, "approval_request");
+
+		verify(config).getTemplate("email.approval_request.ftl");
+		verify(approvalStatusDmRepository).findById(201L);
+	}
+
+	@Test
+	public void shouldSendEmailForBookingApprovalWithEmptyStringsWhenContentOfObjectsInBookingRevisionIsEmpty() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException, MessagingException {
+
+		MailRequest emailRequest = createMailRequest();
+		BookingRevision bookingRevision = createBookingRevision(true);
+		ReflectionTestUtils.setField(emailServiceImpl, "baseUrl", "https://imaginagtion.com/bookingSummary?bookingId={bookingId}&status={status}");
+		
+		when(config.getTemplate("email.approval_request.ftl")).thenReturn(template);
+		when(approvalStatusDmRepository.findById(201L)).thenReturn(Optional.empty());
+		
+		emailServiceImpl.sendEmailForBookingApproval(emailRequest, bookingRevision, "approval_request");
+
+		verify(config).getTemplate("email.approval_request.ftl");
+		verify(approvalStatusDmRepository).findById(201L);
+	}
+
+	@Test
+	public void shouldSendEmailForBookingApprovalWithEmptyStringsWhenObjectsInBookingRevisionAreNull() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException, MessagingException {
+
+		MailRequest emailRequest = createMailRequest();
+		BookingRevision bookingRevision = createBookingRevisionWithNullObjects();
+		ReflectionTestUtils.setField(emailServiceImpl, "baseUrl", "https://imaginagtion.com/bookingSummary?bookingId={bookingId}&status={status}");
+		
+		when(config.getTemplate("email.approval_request.ftl")).thenReturn(template);
+		when(approvalStatusDmRepository.findById(201L)).thenReturn(Optional.empty());
+		
+		emailServiceImpl.sendEmailForBookingApproval(emailRequest, bookingRevision, "approval_request");
+
+		verify(config).getTemplate("email.approval_request.ftl");
+		verify(approvalStatusDmRepository).findById(201L);
+	}
+
+	@Test
+	public void shouldLogExceptionMessageOccuredDuringEmailSending_sendEmailForBookingApproval() {
```

Comment:
test case is just calling send email method on template, as its mocked it wont send any email as i believe... change the test definition to 'should call send email on email template' and verify it.

Reply:
This test is written to cover catch block in a code